UnityProject/Assets/Scripts/Objects/Construction/GrilleObject.cs

```diff
@@ -0,0 +1,113 @@
+	{
+		//start with the default HandApply WillInteract logic.
+		if (!DefaultWillInteract.Default(interaction, side)) return false;
+
+		//only care about interactions targeting us
+		if (interaction.TargetObject != gameObject) return false;
+		//only try to interact if the user has a wrench, screwdriver, metal, or plasteel in their hand
+		if (!Validations.HasItemTrait(interaction.HandObject, CommonTraits.Instance.Screwdriver) &&
+			!Validations.HasItemTrait(interaction.HandObject, CommonTraits.Instance.Wirecutter)) return false;
+		return true;
+	}
+
+	public void ServerPerformInteraction(HandApply interaction)
+	{
+		if (interaction.TargetObject != gameObject) return;
+
+		else if (Validations.HasItemTrait(interaction.HandObject, CommonTraits.Instance.Screwdriver))
+		{
+			if (objectBehaviour.IsPushable)
+			{
+				//secure it if there's floor
+				if (MatrixManager.IsSpaceAt(registerObject.WorldPositionServer, true))
+				{
+					Chat.AddExamineMsg(interaction.Performer, "A floor must be present to secure the grille!");
+					return;
+				}
+
+				if (!ServerValidations.IsAnchorBlocked(interaction))
+				{
+					ToolUtils.ServerUseToolWithActionMessages(interaction, 0.5f,
+						"You start securing the grille...",
+						$"{interaction.Performer.ExpensiveName()} starts securing the grille...",
+						"You secure the grille.",
+						$"{interaction.Performer.ExpensiveName()} secures the grille.",
+						() => ScrewInPlace(interaction));
+					SoundManager.PlayNetworkedAtPos("screwdriver2", gameObject.TileWorldPosition().To3Int(), Random.Range(0.9f, 1.1f));
+				}
+			}
+			else
+			{
+				//unsecure it
+				ToolUtils.ServerUseToolWithActionMessages(interaction, 0.5f,
+					"You start unsecuring the grille...",
+					$"{interaction.Performer.ExpensiveName()} starts unsecuring the grille...",
+					"You unsecure the grille.",
+					$"{interaction.Performer.ExpensiveName()} unsecures the grille.",
+					() => objectBehaviour.ServerSetAnchored(false, interaction.Performer));
+				SoundManager.PlayNetworkedAtPos("screwdriver2", gameObject.TileWorldPosition().To3Int(), Random.Range(0.9f, 1.1f));
+			}
+
+		}
+		else if (Validations.HasItemTrait(interaction.HandObject, CommonTraits.Instance.Wirecutter))
+		{
+
+				ToolUtils.ServerUseToolWithActionMessages(interaction, 1f,
+					"You start to disassemble the grille...",
+					$"{interaction.Performer.ExpensiveName()} starts to disassemble the grille...",
+					"You disassemble the grille.",
+					$"{interaction.Performer.ExpensiveName()} disassembles the grille.",
+					() => Disassemble(interaction));
+			SoundManager.PlayNetworkedAtPos("wirecutter", gameObject.TileWorldPosition().To3Int(), Random.Range(0.9f, 1.1f));
+		}
+	}
+
+	[Server]
+	private void Disassemble(HandApply interaction)
+	{
+		Spawn.ServerPrefab(CommonPrefabs.Instance.MetalRod, registerObject.WorldPositionServer, count: 2);
+		GetComponent<CustomNetTransform>().DisappearFromWorldServer();
```

Comment:
oh wait, replace all occasions of DisappearFromWorldServer in similar construction classes with 
`Despawn.ServerSingle(gameObject);`
otherwise these will keep existing at HiddenPos